src/ui/components/Location/component.ts

```diff
@@ -171,31 +170,55 @@ export default class LocationComponent extends Component {
   async loadFromAPI(locationSignature: RecordSignature) {
     let { store } = this.args;
     try {
-      let location: Location = await store.query((q) =>
+      this.location = await store.query((q) =>
         q.findRecord(locationSignature)
       );
-      let measurements: Measurement[] = await store.query((q) =>
+    } catch {
+      // Fail silently
+    }
```

Comment:
Hm, these changes seem unrelated?

Reply:
I had to separate the `try`s because now we care about which API function fails. If fetching the location fails, we just use whatever we had in the cache. However, if the measurements fetch fails, then we have to do something special: restoring the cache that we ordered to be deleted.

Reply:
Is that still true though? I don't think we still restore deleted records?